src/pcapng-parser.js

```diff
@@ -0,0 +1,199 @@
+		}
+
+		return interfaceDescription;
+	}
+
+	#parseEnhancedPacketBlock() {
+		const blockStart = this.#stream.pos();
+		const magic = this.#readUInt32();
+
+		if (magic !== BLOCK_TYPE_ENHANCED_PACKET) {
+			const expected = BLOCK_TYPE_ENHANCED_PACKET.toString(16).toLocaleUpperCase();
+			const magicHex = magic.toString(16).toLocaleUpperCase();
+			throw new Error(`Invalid PCAP magic. Expected 0x${expected}, got 0x${magicHex}`);
+		}
+
+		const blockLength = this.#readUInt32();
+
+		const enhancedPacket = {
+			interfaceID: this.#readUInt32(),
+			timestampHigh: this.#readUInt32(),
+			timestampLow: this.#readUInt32(),
+			storedLength: this.#readUInt32(),
+			realLength: this.#readUInt32()
+		};
+
+		const interfaceDescription = this.#currentSection.interfaces[enhancedPacket.interfaceID];
+		let interfaceDataLength = 0;
+
+		switch (interfaceDescription.linkLayerType) {
+		case LINKTYPE_ETHERNET:
+			enhancedPacket.interface = this.#parseInterfaceEthernet();
+			interfaceDataLength = 14;
+			break;
+
+		default:
+			throw new Error(`Unsupported interface type 0x${interfaceDescription.linkLayerType.toString(16).toLocaleUpperCase()}`);
+		}
+
+		enhancedPacket.data = this.#stream.readBytes(enhancedPacket.storedLength - interfaceDataLength);
+
+		// * There's padding here to align the data to a
+		// * 32-bit boundary. However due to the way this
+		// * skips the optional data, the padding doesn't
+		// * need to be handled
+
+		this.#stream.seek((blockStart + blockLength) - 4); // * Skip optional data
+
+		const blockLength2 = this.#readUInt32();
+
+		if (blockLength !== blockLength2) {
+			throw new Error(`Invalid trailing block length. Expected ${blockLength}, got ${blockLength2}`);
+		}
+
+		return enhancedPacket;
+	}
+
+	// * INTERFACE PARSERS
+
+	#parseInterfaceEthernet() {
+		return {
+			type: LINKTYPE_ETHERNET,
+			data: {
+				destinationMAC: this.#stream.readBytes(6).toString('hex'),
+				sourceMAC: this.#stream.readBytes(6).toString('hex'),
+				type: this.#stream.readUInt8() * 256 + this.#stream.readUInt8()
+			}
+		};
+	}
+
+	*packets() {
+		this.#stream.seek(0); // * Make sure we're always at the start
+
+		while (this.#stream.hasDataLeft()) {
+			const magic = this.#readUInt32();
+
+			// * Skip back so the block parsers can have the right start position.
+			// * Makes the block parsers follow the docs layout better rather than
+			// * skipping the magic read
+			this.#stream.skip(-4);
+
+			switch (magic) {
+			case BLOCK_TYPE_SECTION_HEADER:
+				this.#currentSection = this.#parseSectionHeaderBlock();
+				break;
+
+			case BLOCK_TYPE_INTERFACE_DESCRIPTION:
+				this.#currentSection.interfaces.push(this.#parseInterfaceDescriptionBlock());
+				break;
+
+			case BLOCK_TYPE_ENHANCED_PACKET:
+				yield this.#parseEnhancedPacketBlock();
+				break;
+
+			default:
+				throw new Error(`Unsupported block type 0x${magic.toString(16).toLocaleUpperCase()}`);
+			}
```

Comment:
According to https://www.ietf.org/staging/draft-tuexen-opsawg-pcapng-02.html#name-block-definition , there are more block types like 0x5. If we don't support this, opening a network dump will throw an error

Reply:
Can you send me the dump you have which uses block type 5? The reason no other blocks are implemented is because I don't have dumps which use them so I can't test